File: libray-managment-app/src/main/java/com/hexad/service/LibraryService.java
package com.hexad.service;

import com.hexad.entity.Library;

public interface LibraryService {

	/**
	 * Load library.
	 *
	 * @param userName the user name
	 * @return the library
	 */
	Library loadLibrary(String userName);
	
	/**
	 * Borrow book.
	 *
	 * @param userName the user name
	 * @param bookId the book id
	 * @return the library
	 */
	Library borrowBook(String userName, Integer bookId);
	
	/**
	 * Submit book.
	 *
	 * @param userName the user name
	 * @param bookId the book id
	 * @return the library
	 */
	Library submitBook(String userName, Integer bookId);
	
}

File: library-managment-frontend/library-managment/src/app/services/library.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Book, LibraryResponse } from '../data-objects/Book';


@Injectable({
  providedIn: 'root'
})
export class LibraryService {
  bookList: Array<Book> = [];
  borrowList : any[];
  userName:string;
  constructor(private httpClient: HttpClient) { }

  loadLibrary(userName : string){
    return this.httpClient.get<LibraryResponse>('http://localhost:8080/library' + '/' + userName);
  }

  borrowBook(userName : string, bookID : number){
    return this.httpClient.post<LibraryResponse>('http://localhost:8080/borrow/' + userName + '/' + bookID,null);
  }

  submitBook(userName : string, bookID : number){
    return this.httpClient.post<LibraryResponse>('http://localhost:8080/submit/' + userName + '/' + bookID,null);
  }
  
}
